src/libmongoc/doc/mongoc_auto_encryption_opts_set_key_expiration.rst

```diff
@@ -0,0 +1,27 @@
+:man_page: mongoc_auto_encryption_opts_set_key_expiration
```

Comment:
Add `mongoc_auto_encryption_opts_set_key_expiration` to the `toctree` of `mongoc_auto_encryption_opts_t.rst` to include in the docs. Fixes error from the docs build:

```
mongoc_auto_encryption_opts_set_key_expiration.rst:document isn't included in any toctree
```

CONTRIBUTING.md documents how to build docs locally. To build libmongoc docs, I use:
```bash
DIR=src/libmongoc/doc
OUTDIR=cmake-build/src/libmongoc/doc/html
poetry run sphinx-autobuild -b html $DIR $OUTDIR --re-ignore ".*.pickle" --re-ignore ".*.doctree" -j auto --port 9000 -W
```